platform/darwin/src/NSDate+MGLAdditions.h

```diff
@@ -7,4 +7,7 @@
 /// Converts from a duration in seconds to a duration object usable in mbgl.
 mbgl::Duration MGLDurationInSeconds(NSTimeInterval duration);
 
+/// Converts from an mbgl duration object to a duration in seconds.
+NSTimeInterval MGLSecondsFromDuration(mbgl::Duration duration);
```

Comment:
No, I don’t think that’s an issue with `std::chrono::duration` or `mbgl::Duration` (which is actually in nanoseconds, my bad). `MGLDurationInSeconds()` performs a [`duration_cast`](http://en.cppreference.com/w/cpp/chrono/duration/duration_cast) to ensure that the output has the right quantity (if not the same unit as the input).